### PR TITLE
New writer oriented rework of docs/README

### PR DESCRIPTION
**Pull request by Ordinant**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -100,46 +114,70 @@ or on Linux with
...
+  Slack](https://trino.io/slack.html).
+* For a larger contribution, create a GitHub pull request as described
+  in [GitHub
+  documentation](https://docs.github.com/en/github/getting-started-with-github).
+  In brief, this means:
+  * [Create a fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) of the
+    [trinodb/trino](https://github.com/trinodb/trino) repository.
+  * Create a working branch in your fork.
+  * Make your edits in your working branch.
+  * In a browser, open your fork in GitHub, which offers to submit a pull
```
Comment (mosabua) [inline docs/README.md:182]: Can users add the "Docs" label?

File: docs/README.md [hunk 2]
```diff
@@ -100,46 +114,70 @@ or on Linux with
...
+To contribute corrections or new explanations to the Trino documentation requires
+only a willingness to help and submission of your [Contributor License
+Agreement](https://github.com/trinodb/cla).
+
+## Workflow
+
+The procedure to add a documentation contribution is the same as for [a code
+contribution](https://trino.io/development/process.html).
+
+* Discuss a small change in the #docs channel of the [Trino
```
Comment (mosabua) [inline docs/README.md:172]: We should add a link to the docs label filter and mention the good first issue label (maybe two links?)

File: docs/README.md [hunk 3]
```diff
@@ -39,7 +39,8 @@ Other useful resources:
 
 ## Tools
 
-The default build of the docs is performed with Apache Maven.
+The default build of the docs is performed with Apache Maven, which requires an
+installation of Java.
```
Comment (mosabua) [inline docs/README.md:43]: It also needs docker.. so maybe we should rearrange this section 

docker (for just authoring)
Jdk and Maven (for full build and validation)

**Review by mosabua — CHANGES_REQUESTED** [review]
Great overall. Made a couple of suggestions .. ping me for re-review after

File: docs/README.md (hunk 3 above)
Comment (Ordinant) [reply docs/README.md:43]: Reworked as suggested.

File: docs/README.md (hunk 2 above)
Comment (Ordinant) [reply docs/README.md:172]: Added two links.

File: docs/README.md (hunk 1 above)
Comment (Ordinant) [reply docs/README.md:182]: With CLA in place, I just successfully added ``docs`` to [7683](https://github.com/trinodb/trino/issues/7683). So I assume that CLA is all that is required. 

According I added ``docs`` label to the Workflow.

File: docs/README.md [hunk 4]
```diff
@@ -39,52 +39,69 @@ Other useful resources:
...
 docs/build
 ```
 
-Sphinx will attempt to perform an incremental build, but it does not work
+Sphinx attempts to perform an incremental build, but this does not work
 in all cases, such as after editing the CSS. You can force a full rebuild
 by doing a Maven clean first:
 
 ```bash
 ./mvnw -pl docs clean
```
Comment (mosabua) [inline docs/README.md:76]: maybe remove this and just say to delete the target directory

Comment (Ordinant) [reply docs/README.md:76]: that's faster, too.

Comment (mosabua) [reply docs/README.md:76]: or do a `rm -rf docs/target` in the example

File: docs/README.md [hunk 5]
```diff
@@ -39,52 +39,69 @@ Other useful resources:
...
+./docs/build
```
Comment (mosabua) [inline docs/README.md:77]: ./ is not needed I think

File: docs/README.md [hunk 6]
```diff
@@ -100,46 +117,85 @@ or on Linux with
...
 
 ```bash
 chromium-browser docs/target/html/index.html
 ```
 
-Alternatively, you can start a web server with that folder as root, e.g. again
-with Python and then open [http://localhost:4000](http://localhost:4000) in a
-web browser.
+Alternatively, you can start a web server with that folder as root, such as
+with the following Python command. You can then open [http://localhost:4000](http://localhost:4000)
```
Comment (mosabua) [inline docs/README.md:128]: wrap?

File: docs/README.md [hunk 7]
```diff
@@ -100,46 +117,85 @@ or on Linux with
...
+## Workflow
+
+The procedure to add a documentation contribution is the same as for [a code
+contribution](https://trino.io/development/process.html).
+
+* In the Trino project's [GitHub Issues
+  list](https://github.com/trinodb/trino/issues), identify documentation issues
+  by filtering on the ``docs`` label.
+
+* If you want to help Trino documentation, but don't know where to start, look
```
Comment (mosabua) [inline docs/README.md:179]: Add the links for these filters from the slack chat

Comment (mosabua) [reply docs/README.md:179]: one for "docs issues" and another "good first doc issues"

**Review by mosabua — CHANGES_REQUESTED** [review]
Some more tweaks to not need Maven ;-)

File: docs/README.md (hunk 5 above)
Comment (Ordinant) [reply docs/README.md:77]: I'm so used to being cd'd to docs, I always type ./build. Fixing.

Comment (Ordinant) [reply docs/README.md:77]: Well, no, we removed that line in favor of rm -rf docs/target/html

File: docs/README.md (hunk 7 above)
Comment (Ordinant) [reply docs/README.md:179]: Confused. You want something added to Trino Slack or here. But if the latter, where to?

Comment (Ordinant) [reply docs/README.md:179]: Figured this out. The links work fine, including preserving the ``highlighting`` inside the link.

**Review by mosabua — APPROVED** [review]
Looks great now.

File: docs/README.md [hunk 8]
```diff
@@ -29,7 +30,7 @@ new documentation:
 - [Present tense](https://developers.google.com/style/tense)
 
 The guidelines include much more material and are used as a guide and enable
```
Comment (m57lyra) [inline docs/README.md:32]: There are a lot of ands in this sentence.

File: docs/README.md [hunk 9]
```diff
@@ -29,7 +30,7 @@ new documentation:
...
-easy decision making. Existing documentation upgrades to follow the guidelines
+easy decision making. Upgrades to existing documentation to follow these guidelines
```
Comment (m57lyra) [inline docs/README.md:33]: Awkward sentence

File: docs/README.md [hunk 10]
```diff
@@ -39,52 +40,68 @@ Other useful resources:
...
+If you have Maven installed and available on the path, you can use the `mvn` command
+directly.
+
+This also performs other checks, and is the authoritative way to build the
+docs. However, using Maven is also somewhat slower than using Sphinx directly.
+
 ## Viewing documentation
 
-However you built the docs, the output HTML files can be found in the folder
+However you build the docs, the generated HTML files can be found in the folder
```
Comment (m57lyra) [inline docs/README.md:104]: ``open`` works on a mac

File: docs/README.md [hunk 11]
```diff
@@ -100,46 +117,102 @@ or on Linux with
...
+  in the Issues list for both the [``docs`` and ``good first issue``
+  labels](https://github.com/trinodb/trino/issues?q=is%3Aissue+is%3Aopen+label%3Adocs+label%3A%22good+first+issue%22).
+
+* If the doc fix you have in mind does not yet have an issue, add one (which
+  requires a signed CLA). Add the ``docs`` label to your new issue.
+
+* Discuss a small change in the #docs channel of the [Trino
+  Slack](https://trino.io/slack.html).
+
+* For a larger contribution, create a GitHub pull request as described
```
Comment (m57lyra) [inline docs/README.md:190]: This is a little misleading, as any change large or small will eventually need a PR.

File: docs/README.md [hunk 12]
```diff
@@ -100,46 +117,102 @@ or on Linux with
...
+  in [GitHub
+  documentation](https://docs.github.com/en/github/getting-started-with-github).
+  In brief, this means:
+
+  * [Create a fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) of the
+    [trinodb/trino](https://github.com/trinodb/trino) repository.
+
+  * Create a working branch in your fork.
+
+  * Make your edits in your working branch.
```
Comment (m57lyra) [inline docs/README.md:200]: branch, and push them to your fork.

**Review by m57lyra — APPROVED** [review]
LGTM

**Review by electrum — APPROVED** [review]

File: docs/README.md (hunk 8 above)
Comment (Ordinant) [reply docs/README.md:32]: Original content that I didn't dare change. I'll try to de-and it a bit.

Comment (Ordinant) [reply docs/README.md:32]: Fixed, after a fashion

File: docs/README.md (hunk 9 above)
Comment (Ordinant) [reply docs/README.md:33]: somewhat less awkward

File: docs/README.md (hunk 11 above)
Comment (Ordinant) [reply docs/README.md:190]: Fixed.

File: docs/README.md (hunk 12 above)
Comment (Ordinant) [reply docs/README.md:200]: fixed as suggested

File: docs/README.md [hunk 13]
```diff
@@ -39,52 +40,68 @@ Other useful resources:
...
-## Default build
+The [fast doc build option](#fast-doc-build-option) requires *only* a local
+installation of [Docker Desktop on
+Mac](https://docs.docker.com/docker-for-mac/install/) or [Docker Engine on
+Linux](https://docs.docker.com/engine/install/). No other tools are required.
+
+The default formal build of the docs is performed with Apache Maven, which requires an
+installation of a Java Development Kit.
+
+Python 3 is required for one optional way to view the generated documentation.
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/README.md:58]: Remove only the Python requirement line? If so, done. If you mean to remove more, discuss.

**Pull request merged**